src/NBitcoin/Block.cs

```diff
@@ -252,12 +258,11 @@ public Block(byte[] bytes)
             this.ReadWrite(bytes);
         }
 
-
         public void ReadWrite(BitcoinStream stream)
         {
             stream.ReadWrite(ref this.header);
             stream.ReadWrite(ref this.transactions);
-            if (Block.BlockSignature)
+            if ((stream.TransactionOptions ?? NetworkOptions.TemporaryOptions).IsProofOfStake)
```

Comment:
Same here lets try to set the `stream.TransactionOptions` to default in `BitcoinStream` and not play around with checking for null then later its easier to change.

Reply:
It's already being defaulted on the stream object. Will remove this check.